src/applications/facility-locator/containers/VAMap.jsx

```diff
@@ -128,25 +137,38 @@ class VAMap extends Component {
   }
 
   componentWillUnmount() {
-    // call the func returned by browserHistory.listen to unbound the listener
+    // call the func returned by browserHistory.listen to unbind the listener
     this.listener();
   }
 
-  syncStateWithLocation(location) {
+  /**
+   * Presumably handles the case if a user manually types a change to the
+   * address bar and thereby updates the location as tracked by ReactRouter?
+   * (i.e. route changes not handled through the Router)
+   * 
+   * @param {Object} location ReactRouter location object
+   */
+  syncStateWithLocation = (location) => {
     if (
-      location.query.address &&
-      this.props.currentQuery.searchString !== location.query.address && !this.props.currentQuery.inProgress
+      location.query.address
+      && this.props.currentQuery.searchString !== location.query.address
+      && !this.props.currentQuery.inProgress
     ) {
-      this.props.searchWithAddress({
+      this.props.genBBoxFromAddress({
         searchString: location.query.address,
         context: location.query.context,
       });
     }
   }
 
-  // pushes coordinates to URL so that map link is useful for sharing
-  // TODO (bshyong): try out existing query-string npm library
+  /**
+   * Regenerates the URL based on the given parameters so that
+   * the map link stays useful for sharing. 
+   * 
+   * @param {Object} params 
+   */
   updateUrlParams = (params) => {
+    // TODO (bshyong): try out existing query-string npm library
```

Comment:
What is this todo for?

Reply:
I would imagine we would have to ask 'bshyong'?

I'm honestly not sure, it was part of the codebase when I started on it. Although now that I think about it, it may be something along the lines of what @U-DON said above for API things:
> Maybe consider using the `append-query` library that we have available in our dependencies.

(Some idea to pull in a package and do a different/better job of handling URL params?)

Reply:
Yea, this is more of the function:

```jsx
const queryParams = compact(map({
  ...location.query,
  zoomLevel: currentQuery.zoomLevel,
  page: currentQuery.currentPage,
  address: currentQuery.searchString,
  facilityType: currentQuery.facilityType,
  serviceType: currentQuery.serviceType,
  ...params,
}, (value, key) => {
  if (value) { return `${key}=${value}`; }
  return null;
})).join('&');
```

Handwritten logic to handle building out the URL query parameters. It's sounding like a "I should come back and do this less manually later with the library in the comment". But can't be 100% sure.